### ISSUE TITLE
Keras callback stops receiving some params with tensorflow 2.2.0

### ISSUE THREAD

**Issue by Lothiraldan** [tensorflow/tensorflow#40515]
**System information**
- Have I written custom code (as opposed to using a stock example script provided in TensorFlow): Yes
- OS Platform and Distribution (e.g., Linux Ubuntu 16.04): Linux Fedora 31
- TensorFlow installed from (source or binary): binary
- TensorFlow version (use command below): v2.2.0-rc4-8-g2b96f3662b 2.2.0
- Python version: Python 3.7.5
- CUDA/cuDNN version: No GPU
- GPU model and memory: No GPU

**Describe the current behavior**

When defining a custom Keras callback, the `set_params` method receives a subset of parameters in tensorflow 2.2.0 in comparison to what it used to receive in tensorflow 2.1.1.

In tensorflow 2.2.0, it receives:

```
{'verbose': 1, 'epochs': 2, 'steps': 1}
```

**Describe the expected behavior**

I would expect to get the same params as with tensorflow 2.1.1, which are:

```
{'batch_size': 4, 'epochs': 2, 'steps': None, 'samples': 4, 'verbose': 1, 'do_validation': False, 'metrics': ['loss']}
```

**Standalone code to reproduce the issue**

I used the following script to test the behavior with the two version of tensorflows and also trying with `keras` versus `tf.keras`:

```
import numpy as np
# import keras
import tensorflow.keras as keras

def build_xor_data():
    x_train = [np.array([[0, 0], [0, 1], [1, 0], [1, 1]], dtype=float)]
    y_train = [np.array([[0], [1], [1], [0]], dtype=float)]

    return x_train, y_train


def build_xor_model_keras():
    input_layer = keras.layers.Input(shape=(2,))
    hidden_layer = keras.layers.Dense(2, activation="sigmoid")(input_layer)
    output_layer = keras.layers.Dense(1, activation="sigmoid")(hidden_layer)
    model = keras.models.Model(inputs=input_layer, outputs=output_layer)
    model.compile(loss="mse", optimizer="adam")
    return model

x_train, y_train = build_xor_data()

model = build_xor_model_keras()

class DebugCallback(keras.callbacks.Callback):

    def set_params(self, params):
        print("SET PARAMS", locals())

print("KERAS VERSION", keras.__version__)
model.fit(x_train, y_train, batch_size=4, epochs=2, callbacks=[DebugCallback()])
```

I got the following outputs:

|          	| keras                                                                                                                  	| tf.keras                                                                                                            	|
|----------	|------------------------------------------------------------------------------------------------------------------------	|---------------------------------------------------------------------------------------------------------------------	|
| tf 2.1.1 	| {'batch_size': 4, 'epochs': 2, 'steps': None, 'samples': 4, 'verbose': 1, 'do_validation': False, 'metrics': ['loss']} 	| {'batch_size': 4, 'epochs': 2, 'steps': 1, 'samples': 4, 'verbose': 0, 'do_validation': False, 'metrics': ['loss']} 	|
| tf 2.2.0 	| {'batch_size': 4, 'epochs': 2, 'steps': None, 'samples': 4, 'verbose': 1, 'do_validation': False, 'metrics': ['loss']} 	| {'verbose': 1, 'epochs': 2, 'steps': 1}                                                                             	|

In tensorflow 2.2.0, we don't get anymore the following parameters:
* batch_size
* samples
* do_validation
* metrics

After reading the release note, section `Breaking changes` https://github.com/tensorflow/tensorflow/releases/tag/v2.2.0, I could expect to not receive `metrics` anymore. Are the other missing params also linked to that breaking change?

**Comment by Saduf2019** [comment]
I am able to replicate this issue, please find the [gist here](https://colab.research.google.com/gist/Saduf2019/7277eb9f69678632d428fb4df62bfb2f/untitled231.ipynb).

**Comment by Lothiraldan** [comment]
Let me know if you need more information from my side.

**Comment by saikumarchalla** [comment]
Was able to reproduce the issue using TF [2.1](https://colab.research.google.com/gist/saikumarchalla/7f581eec506edfcf4e9887905bcc657a/untitled231.ipynb) and [2.5](https://colab.research.google.com/gist/saikumarchalla/10b710439c0d2301b45591b87d776e36/untitled84.ipynb). Please find the attached gist.Thanks!

**Comment by sushreebarsa** [comment]
@Lothiraldan This issue is still replicating in  TF v2.7.0. Please find the  [gist](https://colab.research.google.com/gist/sushreebarsa/cdb73d0c2ce1138dcf0970aae9b6c338/untitled84.ipynb) for reference .Thanks!

**Comment by tensorflowbutler** [comment]
Hi There,

 This is a stale issue. As you are using an older version of tensorflow, we are checking to see if you still need help on this issue. Please test the issue with the latest TensorFlow (TF2.7 and tf-nightly). If the issue still persists with the newer versions of TF, please feel free to open it in [keras-team/keras](https://github.com/keras-team/keras/issues) repository by providing details about the issue and a standalone code to reproduce the issue. Thanks! 

 Please note that Keras development has moved to a separate Keras-team/keras repository to focus entirely on only Keras. Thanks!

**Comment by kshitij12345** [comment]
Still the same on 2.8
